internal/appfunction/context.go

```diff
@@ -269,3 +273,49 @@ func (appContext *Context) PipelineId() string {
 	id, _ := appContext.GetValue(interfaces.PIPELINEID)
 	return id
 }
+
+// Publish pushes data to the messagebus
+func (appContext *Context) Publish(v any) error {
+	messageClient := bootstrapContainer.MessagingClientFrom(appContext.Dic.Get)
+	if messageClient == nil {
+		return fmt.Errorf("messagebus is disabled via configuration")
+	}
+
+	triggerConfig := container.ConfigurationFrom(appContext.Dic.Get).Trigger
+	baseTopic := bootstrapContainer.ConfigurationFrom(appContext.Dic.Get).GetBootstrap().MessageBus.BaseTopicPrefix
+
+	payload, err := json.Marshal(v)
+	if err != nil {
+		return fmt.Errorf("failed to marshal data: %v", err)
+	}
+	message := types.NewMessageEnvelope(payload, context.Background())
+	err = messageClient.Publish(message, common.BuildTopic(baseTopic, triggerConfig.PublishTopic))
+	if err != nil {
+		return fmt.Errorf("failed to publish data to messagebus: %v", err)
+	}
+	return nil
+}
+
+// Publish pushes data to the messagebus for a given topic
+func (appContext *Context) PublishWithTopic(topic string, data any) error {
+	messageClient := bootstrapContainer.MessagingClientFrom(appContext.Dic.Get)
+	if messageClient == nil {
+		return fmt.Errorf("messagebus is disabled via configuration")
+	}
+
+	config := bootstrapContainer.ConfigurationFrom(appContext.Dic.Get)
+
+	full_topic := common.BuildTopic(config.GetBootstrap().MessageBus.BaseTopicPrefix, topic)
+	payload, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("failed to marshal data: %v", err)
```

Comment:
create constant for the repeated error message
```suggestion
		return fmt.Errorf("failed to marshal data for publishing: %v", err)
```